creating_packages/define_abi_compatibility.rst

```diff
@@ -701,6 +701,48 @@ generated with the ``recipe_revision_mode`` can be resolved if no package for th
             self.compatible_packages.append(p)
 
 
+.. _full_transitive_package_id:
+
+Enabling full transitivity in package_id modes
+++++++++++++++++++++++++++++++++++++++++++++++
+
+.. warning::
+
+    This will become the default behavior in the future (Conan 2.0). It is recommended to activate it when possible (it might require rebuilding some packages,
+    as their package IDs will change)
+
+
+When a package declares in its ``package_id()`` method that it is not affected by its dependencies, that will propagate down
+to the indirect consumers of that package. There are several ways this can be done, ``self.info.header_only()``, ``self.info.requires.clear()``,
+``self.info.requires.remove["dep"]`` and ``self.info.requires.unrelated_mode()``, for example.
+
+Let's assume for the discussion that it is a header only library, using the ``self.info.header_only()`` helper. This header only package has
+a single dependency, which is a static library. Then, downstream
+consumers of the header only library that uses a package mode different from the default, should be also affected by the upstream
+transitivity dependency. Lets say that we have the following scenario:
+
+- ``App/1.0`` depends on ``PkgC/1.0`` and ``PkgA/1.0``
+- ``PkgC/1.0`` depends only on ``PkgB/1.0``
+- ``PkgB/1.0`` depends on ``PkgA/1.0``, and defines ``self.info.header_only()`` in its ``package_id()``
+- We are using ``full_version_mode``
```

Comment:
I would move this to the line 722: "Lets say that we have the following scenario where we are using full_version_mode:"